mesonbuild/backend/backends.py

```diff
@@ -709,15 +709,15 @@ def generate_basic_compiler_args(self, target: build.BuildTarget, compiler: 'Com
         commands += compiler.get_buildtype_args(self.get_option_for_target(OptionKey('buildtype'), target))
         commands += compiler.get_optimization_args(self.get_option_for_target(OptionKey('optimization'), target))
         commands += compiler.get_debug_args(self.get_option_for_target(OptionKey('debug'), target))
-        # MSVC debug builds have /ZI argument by default and /Zi is added with debug flag
-        # /ZI needs to be removed in that case to avoid cl's warning to that effect (D9025 : overriding '/ZI' with '/Zi')
-        if ('/ZI' in commands) and ('/Zi' in commands):
-            commands.remove('/Zi')
         # Add compile args added using add_project_arguments()
         commands += self.build.get_project_args(compiler, target.subproject, target.for_machine)
         # Add compile args added using add_global_arguments()
         # These override per-project arguments
         commands += self.build.get_global_args(compiler, target.for_machine)
+        # MSVC debug builds have /ZI argument by default and /Zi is added with debug flag
+        # /ZI needs to be removed in that case to avoid cl's warning to that effect (D9025 : overriding '/ZI' with '/Zi')
```

Comment:
This comment is no longer up to date as `/ZI` is not added by default. Instead the comment should say something like "Using both /ZI and /Zi at the same times produces a compiler warning, We don't add /ZI by default so if it is being used it is because the user has explicitly enabled it so it should override /Zi.".

Reply:
Done